src/WebJobs.Script/Workers/Rpc/WebHostRpcWorkerChannelManager.cs

```diff
@@ -57,6 +58,16 @@ public WebHostRpcWorkerChannelManager(IScriptEventManager eventManager,
             _applicationHostOptions = applicationHostOptions;
             _hostingConfigOptions = hostingConfigOptions;
 
+            languageWorkerOptions.OnChange(async languageWorkerOptions =>
+            {
+                IRpcWorkerChannel rpcWorkerChannel = await GetChannelAsync(_workerRuntime);
+                if (rpcWorkerChannel != null && !UsePlaceholderChannel(rpcWorkerChannel))
+                {
+                    _logger.LogInformation("Language worker options changed, and the placeholder worker channel is invalid for other reasons. Shutting down the channel.");
+                    await ShutdownChannelIfExistsAsync(_workerRuntime, rpcWorkerChannel.Id);
```

Comment:
What's the scenario here, and why didn't we have this behavior before?

Reply:
Will need to review this to comment. That's part of a set of changes pushed by @kshyju .

Changes to the signal used to reload `LanguageWorkerOptions` were required. I'm concerned that the logic here is assuming this will always change as a result of specialization, which may change in the future.

At the very least, the message needs to be a bit more descriptive.